contracts/automation/andromeda-evaluation/src/contract.rs

```diff
@@ -0,0 +1,1045 @@
+    }
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    match msg {
+        ExecuteMsg::AndrReceive(msg) => contract.execute(deps, env, info, msg, execute),
+        ExecuteMsg::Evaluate {
+            user_value,
+            operation,
+        } => execute_evaluate(deps, env, info, user_value, operation),
+        ExecuteMsg::ChangeConditionAddress { address } => {
+            execute_change_condition_address(deps, env, info, address)
+        }
+        ExecuteMsg::ChangeQueryAddress { address } => {
+            execute_change_query_address(deps, env, info, address)
+        }
+    }
+}
+
+fn execute_change_query_address(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    address: AndrAddress,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    // Only the contract's owner can update the Execute ADO address
+    ensure!(
+        ADOContract::default().is_owner_or_operator(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {}
+    );
+    QUERY_ADO_ADDRESS.save(deps.storage, &address)?;
+    Ok(Response::new().add_attribute("action", "changed_query_ado_address"))
+}
+
+fn execute_change_condition_address(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    address: AndrAddress,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    // Only the contract's owner can update the Execute ADO address
+    ensure!(
+        ADOContract::default().is_owner_or_operator(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {}
+    );
+    CONDITION_ADO_ADDRESS.save(deps.storage, &address)?;
+    Ok(Response::new().add_attribute("action", "changed_CONDITION_ADO_ADDRESS"))
+}
+
+fn execute_evaluate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    user_value: Uint128,
+    operation: Operators,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+
+    let contract = ADOContract::default();
+    let app_contract = contract.get_app_contract(deps.storage)?;
+
+    // get the address of the ADO that will interpret our result
+    let contract_addr = CONDITION_ADO_ADDRESS.load(deps.storage)?.get_address(
+        deps.api,
+        &deps.querier,
+        app_contract.clone(),
+    )?;
+
+    // get the address of the oracle contract that will provide data to be compared with the user's data
+    let query_addr =
+        QUERY_ADO_ADDRESS
+            .load(deps.storage)?
+            .get_address(deps.api, &deps.querier, app_contract)?;
+
+    let oracle_value: Uint128 = deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+        contract_addr: query_addr,
+        msg: to_binary(&andromeda_automation::counter::QueryMsg::Count {})?,
+    }))?;
+
+    match operation {
+        Operators::Greater => Ok(Response::new()
```

Comment:
Bit of redundancy here, the oracle/user value comparator result can be stored in a variable and the message can be compiled separately